internal/api/approval/approval.go

```diff
@@ -387,24 +387,25 @@ func getTeamNameFromTeamHandle(teams []*github.Team, v string) (string, error) {
 }
 
 func (a *Approval) allowedAndIgnoreReviewers(ctx context.Context, pr *PR, members []*github.User, ignoreContributors bool) ([]string, []string, error) {
-	if !ignoreContributors {
-		var allowedMembers []string
-		for _, m := range members {
-			allowedMembers = append(allowedMembers, m.GetLogin())
+	commits := []*github.RepositoryCommit{}
+	if ignoreContributors {
+		var err error
+		commits, err = a.client.GetPRCommits(ctx, pr.OwnerLogin, pr.RepoName, pr.Number)
+		if err != nil {
+			return nil, nil, err
 		}
-		return allowedMembers, []string{}, nil
 	}
 
-	commits, err := a.client.GetPRCommits(ctx, pr.OwnerLogin, pr.RepoName, pr.Number)
+	events, err := a.client.GetIssuesEvents(ctx, pr.OwnerLogin, pr.RepoName, pr.Number)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	allowed, ignored := filterAllowedAndIgnoreReviewers(members, commits)
+	allowed, ignored := filterAllowedAndIgnoreReviewers(members, commits, events)
```

Comment:
Now that approvals can be rejected for two reasons (being contributor or having reopened the pull request), I think it would be beneficial to refactor `ReportIgnoredReviews` function.

You could make that function more robust and differentiate between different reasons for rejecting the approval. But the easiest way is to adjust the language there to make it more broad.

Otherwise, the message added by GTA can be confusing if approval is rejected due to being added by someone who reopened the pull request.